common/djangoapps/enrollment/data.py

```diff
@@ -272,7 +287,13 @@ def _get_user(user_id):
 def _update_enrollment(enrollment, is_active=None, mode=None):
     enrollment.update_enrollment(is_active=is_active, mode=mode)
     enrollment.save()
-    return CourseEnrollmentSerializer(enrollment).data
+    # Edraak: use EdraakCourseEnrollmentSerializer to serialize enrollments
+    data = EdraakCourseEnrollmentSerializer(
+        enrollment,
+        context={'request': get_request_or_stub()}
```

Comment:
```suggestion
data = EdraakCourseEnrollmentSerializer(enrollment).data
```

More succinct. Please check the other comments.

Reply:
@OmarEdraak `context` is needed by `CourseDetailMarketingSerializer`; it will through errors if removed

`CourseDetailMarketingSerializer` is used here (with the name `EdraakCourseSerializer`) https://github.com/Edraak/edraak-platform/blob/5c56260674a6c804c16a72880b88f303b9d4e586/common/djangoapps/enrollment/serializers.py#L110

Reply:
Even with this @shadinaif https://github.com/Edraak/edraak-platform/pull/98#discussion_r301576935 ??

I don't see why it's needed.

Reply:
Yes, @OmarEdraak . 
I get a `KeyError: 'request'` error in this line:
https://github.com/Edraak/edraak-platform/blob/e79bc9878d3ffcdd8b05373f7a78885a7e0a6323/lms/djangoapps/course_api/serializers.py#L100
And even if I added the request I get the same error
```
edraak_course_details = EdraakCourseSerializer(source="course_overview", context={'request': get_request_or_stub()})
```

Reply:
Got it, thanks @shadinaif! We can keep this one as-is.